addon_imps/storage/dropbox.py

```diff
@@ -0,0 +1,118 @@
+import dataclasses
+import typing
+
+from addon_toolkit.interfaces import storage
+from addon_toolkit.interfaces.storage import ItemType
+
+
+class DropboxStorageImp(storage.StorageAddonHttpRequestorImp):
+    """storage on dropbox.com
+    see https://www.dropbox.com/developers/documentation/http/documentation
+    """
+
+    async def get_external_account_id(self, _: dict[str, str]) -> str:
+        return ""
+
+    async def list_root_items(self, page_cursor: str = "") -> storage.ItemSampleResult:
+        if page_cursor:
+            async with self.network.POST(
+                "files/list_folder/continue",
+                json={"cursor": page_cursor},
+            ) as _response:
+                _parsed = _DropboxParsedJson(await _response.json_content())
+                return storage.ItemSampleResult(
+                    items=list(_parsed.item_results()),
+                    total_count=len(_parsed.response_json["entries"]),
+                    next_sample_cursor=_parsed.response_json["cursor"],
+                )
+        result = await self.list_child_items("")
+        return result
+
+    async def get_item_info(self, item_id: str) -> storage.ItemResult:
+        if not item_id:
+
+            return storage.ItemResult(
+                item_id="",
+                item_name="root folder",
+                item_type=ItemType.FOLDER,
+            )
+        else:
+            async with self.network.POST(
+                "files/get_metadata",
+                json={
+                    "path": _make_item_id(item_id),
+                },
+            ) as _response:
+                _parsed = _DropboxParsedJson(await _response.json_content())
+
+                return _parsed.single_item_result()
+
+    async def list_child_items(
+        self,
+        item_id: str,
+        page_cursor: str = "",
+        item_type: storage.ItemType | None = None,
+    ) -> storage.ItemSampleResult:
+        if page_cursor:
+            async with self.network.POST(
+                "files/list_folder/continue",
+                json={"cursor": page_cursor},
```

Comment:
don't we still need path if we have cursor?

Reply:
Cursor already have information about  needed path